src/program/lwaftr/setup.lua

```diff
@@ -18,23 +19,55 @@ local ipv6_apps  = require("apps.lwaftr.ipv6_apps")
 local ndp        = require("apps.lwaftr.ndp")
 local vlan       = require("apps.vlan.vlan")
 local numa       = require("lib.numa")
+local cltable    = require("lib.cltable")
 local ipv4       = require("lib.protocol.ipv4")
 local ethernet   = require("lib.protocol.ethernet")
 local ipv4_ntop  = require("lib.yang.util").ipv4_ntop
 local S          = require("syscall")
 local engine     = require("core.app")
 
+
+
 local capabilities = {['ietf-softwire']={feature={'binding', 'br'}}}
 require('lib.yang.schema').set_default_capabilities(capabilities)
 
 local function convert_ipv4(addr)
    if addr ~= nil then return ipv4:pton(ipv4_ntop(addr)) end
 end
 
-function lwaftr_app(c, conf)
+-- Checks the existance and NUMA affinity of PCI devices
+-- NB: "nil" can be passed in and will be siliently ignored.
+local function validate_pci_devices(devices)
+   for _, address in pairs(devices) do
+      assert(lwutil.nic_exists(address),
+             ("Could not locate PCI device '%s'"):format(address))
+      numa.check_affinity_for_pci_addresses(address)
```

Comment:
I think this `check_affinity_for_pci_addresses` takes an array of addresses; can you verify?